app/forms/scan_form.rb

```diff
@@ -18,7 +18,7 @@ class ScanForm
     scan.save
   end
 
-  validate :check_available_coordinates, if: proc { |l| l.start_position.present? && l.location.present? }
+  validate :check_available_coordinates, if: proc { |l| l.location.present? && l.location.is_a?(OrderedLocation) }
```

Comment:
Is `l.location.is_a?(OrderedLocation)` the same as` l.location.ordered?`, which is set up by 
create_subclass_methods in Location, and generated by SubclassChecker. Just saying as an alternative in case explicit type checking is to be avoided.